File: backend/app/app_tenant/kb/schemas/kb.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.models.kb import DocumentStatus


class KnowledgeBaseCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=128)
    description: str | None = None
    is_public: bool = False
    chunk_size: int = Field(500, ge=100, le=4000)
    chunk_overlap: int = Field(50, ge=0, le=500)


class KnowledgeBaseUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    is_public: bool | None = None
    chunk_size: int | None = Field(None, ge=100, le=4000)
    chunk_overlap: int | None = Field(None, ge=0, le=500)


class KnowledgeBaseOut(BaseModel):
    id: UUID
    tenant_id: UUID
    name: str
    description: str | None
    is_public: bool
    embedding_dimension: int
    chunk_size: int
    chunk_overlap: int
    created_at: datetime

    model_config = {"from_attributes": True}


class DocumentOut(BaseModel):
    id: UUID
    kb_id: UUID
    tenant_id: UUID
    filename: str
    mime_type: str
    file_size: int
    status: DocumentStatus
    fail_reason: str | None
    celery_task_id: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class SearchRequest(BaseModel):
    query: str = Field(..., min_length=1)
    top_k: int = Field(10, ge=1, le=50)


class SearchHit(BaseModel):
    chunk_id: UUID
    document_id: UUID
    content: str
    score: float
    filename: str | None = None


class SearchResponse(BaseModel):
    query: str
    hits: list[SearchHit]

File: ui/workbench/lib/types/kb.ts
/** 域类型：与 backend OpenAPI 对齐。 */

import type { EnumOption } from "@/lib/enum-meta";
export interface KnowledgeBase {
  id: string;
  tenant_id?: string;
  name: string;
  description?: string | null;
  is_public?: boolean;
  embedding_model_config_id: string;
  embedding_model_name?: string | null;
  visual_embedding_model_config_id?: string | null;
  visual_embedding_model_name?: string | null;
  embedding_dimension: number;
  chunk_size?: number;
  chunk_overlap?: number;
  retrieval_mode?: "vector" | "hybrid";
  hybrid_alpha?: number;
  rerank_model_config_id?: string | null;
  rerank_model_name?: string | null;
  rerank_candidate_k?: number;
  created_at?: string;
}

export interface Document {
  id: string;
  kb_id: string;
  filename: string;
  mime_type: string;
  file_size: number;
  status: string;
  fail_reason?: string | null;
  chunk_count?: number | null;
  created_at: string;
}

export interface DocumentChunk {
  id: string;
  document_id: string;
  kb_id: string;
  chunk_index: number;
  content: string;
  page_no?: number | null;
  created_at: string;
}

export interface KbQuota {
  used_knowledge_bases: number;
  max_knowledge_bases: number;
  used_storage_mb: number;
  max_storage_mb: number;
  max_file_mb: number;
}

export interface KbSearchLog {
  id: string;
  tenant_id: string;
  kb_id: string | null;
  kb_ids: string[] | null;
  query: string;
  top_k: number;
  hit_count: number;
  latency_ms: number;
  retrieval_mode: string;
  source: string;
  actor_user_id: string | null;
  agent_id: string | null;
  created_at: string;
}
